include/mbedtls/config.h

```diff
@@ -1370,6 +1370,15 @@
  */
 #define MBEDTLS_SSL_TRUNCATED_HMAC
 
+/**
+ * \def MBEDTLS_SSL_PREVERIFY_CB
+ *
+ * Enable support for a pre-verification callback for received certificates.
+ *
+ * Uncomment this to enable support for the preverification callback
+ */
+//#define MBEDTLS_SSL_PREVERIFY_CB
```

Comment:
We have far too many features already defined in `config.h`, and I don't think this feature is substantial enough to merit yet another config option of `MBEDTLS_SSL_PREVERIFY_CB`. I'm aware this contradicts what I said earlier. This isn't going into a LTS release, so we can afford to extend the context and break the ABI.

As written the feature is optional because if it's not set, you don't have to call these callbacks. In terms of code size, X.509 is so large, that I don't think a few extra bytes for this callback is worth worrying about. So I think I'd just extend the API with this feature, and not introduce the config option.